Dockerfile

```diff
@@ -1,38 +1,66 @@
-FROM ubuntu:16.04
+FROM ubuntu:latest
 
 ARG BUILDER_UID=9999
 # ARG DOCKER_GID=9999
+ARG DEBIAN_FRONTEND=noninteractive
 
+ENV TZ=Australia/Hobart
 ENV LC_ALL C.UTF-8
 ENV LANG C.UTF-8
 ENV PATH /home/builder/.local/bin:$PATH
+ENV PYTHON_VERSION 3.5.2
 
-RUN apt-get update && apt-get install -y --no-install-recommends \
+RUN apt-get update && \
+    apt-get install -y software-properties-common && \
+    rm -rf /var/lib/apt/lists/*
+
+RUN add-apt-repository ppa:rael-gc/rvm && apt-get update
+
+RUN if [ X"$PYTHON_VERSION" = X"3.5.2" ]; \
+        then apt-get install -y libssl1.0-dev; \
+        else apt-get install -y  libssl-dev; \
+    fi
+
+RUN apt-get install -y --no-install-recommends \
     build-essential \
     ca-certificates \
     git \
     libmagic1 \
     libudunits2-dev \
     python3-dev \
     wget \
-    # docker.io \
+    libffi-dev \
+    # Pyenv pre-requisites
+    make zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev xz-utils tk-dev libffi-dev liblzma-dev python-openssl git \
```

Comment:
This set of packages fails for installing other versions of python, eg 3.8.12. Full set of recommended dependencies at https://github.com/pyenv/pyenv/wiki#suggested-build-environment

Reply:
I don't seem to be hitting any such problem... Can you share the exact error?

I've tested this with 3.5.2, 3.8.10, 3.8.12. Later versions fail due to python dependency issues but I'm not going to waste any time on that now :)